### PR TITLE
Stats: Refactor shortcut matching

### PR DESCRIPTION
**Pull request by a8ck3n**
## Proposed Changes

Cleans up the logic used when matching the date range to a shortcut.

**Note:** Based on work done in #83170 so that should be merged first.

## Testing Instructions

1. Open Live branch to the **Stats → Traffic** page.
2. Enable date picker: `?flags=stats/date-control`
3. Manually enter some dates that should match shortcuts and confirm the shortcut is properly selected.

An easy way to test this is to pick a shortcut from the list, then manually change the start date by one day and update. Confirm the button label shows a full date range and no shortcut is selected. Then change the start date back and update. The label should be set back to the shortcut name and it should be selected in the list.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by kangzj — COMMENTED** [review]
'Today' and 'Yesterday' shortcut got mixed together for some reason 🤔

**Review by kangzj — APPROVED** [review]
👍

File: client/my-sites/stats/stats-date-control/index.tsx [hunk 1]
```diff
@@ -103,35 +103,30 @@ const StatsDateControl = ( { slug, queryParams, dateRange }: StatsDateControlPro
...
-		// Yesterday
-		if ( yesterday === dateRange.chartEnd && dateRange.daysInRange === 1 ) {
-			return shortcutList[ 1 ];
-		}
-		return null;
+		const shortcut = shortcutList.find( ( element ) => {
+			if (
+				yesterday === dateRange.chartEnd &&
+				dateRange.daysInRange === element.range + 1 &&
+				element.id === 'yesterday'
```
Comment (kangzj) [inline client/my-sites/stats/stats-date-control/index.tsx:114]: I fixed it but the logic looks a bit confusing now..

Comment (a8ck3n) [reply client/my-sites/stats/stats-date-control/index.tsx:114]: Were you seeing a bug with the logic here, @kangzj? I'm not sure I understand what the issue is. I'll do some testing with this again, just to be safe.

**Pull request merged**